From 4750d4ce999c04b25574649607d8ca668b8b5443 Mon Sep 17 00:00:00 2001
From: Daan De Meyer <daan.j.demeyer@gmail.com>
Date: Mon, 3 Mar 2025 11:24:58 +0100
Subject: [PATCH] network: Use RTNH_COMPARE_MASK in route_can_update()

Let's only compare flags that can be set by userspace and not all
flags.

Fixes #36544
---
 src/network/networkd-route.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/network/networkd-route.c b/src/network/networkd-route.c
index 9498fa4ccc8f2..cdc1f307af02e 100644
--- a/src/network/networkd-route.c
+++ b/src/network/networkd-route.c
@@ -1413,7 +1413,7 @@ bool route_can_update(const Route *existing, const Route *requesting) {
                         return false;
                 if (existing->type != requesting->type)
                         return false;
-                if (existing->flags != requesting->flags)
+                if ((existing->flags & ~RTNH_COMPARE_MASK) != (requesting->flags & ~RTNH_COMPARE_MASK))
                         return false;
                 if (!in6_addr_equal(&existing->prefsrc.in6, &requesting->prefsrc.in6))
                         return false;
